crates/revm/src/optimism/processor.rs

```diff
@@ -179,10 +179,10 @@ where
                 // receipt hashes should be computed when set. The state transition process ensures
                 // this is only set for post-Canyon deposit transactions.
                 #[cfg(feature = "optimism")]
-                deposit_receipt_version: self
-                    .chain_spec()
-                    .is_fork_active_at_timestamp(Hardfork::Canyon, block.timestamp)
-                    .then_some(1),
+                deposit_receipt_version: (transaction.is_deposit() &&
+                    self.chain_spec()
+                        .is_fork_active_at_timestamp(Hardfork::Canyon, block.timestamp))
+                .then_some(1),
```

Comment:
@clabby this looks right to me, wdyt?

Reply:
This does look correct! Working right now because the receipt version isn't included in the RLP if the `deposit_nonce` isn't set. Better to not set the receipt version if the tx isn't a deposit.